rational.c

```diff
@@ -2467,31 +2467,26 @@ string_to_r_strict(VALUE self, int raise)
 
 /*
  * call-seq:
- *    str.to_r  ->  rational
- *
- * Returns the result of interpreting leading characters in +str+
- * as a rational.  Leading whitespace and extraneous characters
- * past the end of a valid number are ignored.
- * Digit sequences can be separated by an underscore.
- * If there is not a valid number at the start of +str+,
- * zero is returned.  This method never raises an exception.
- *
- *    '  2  '.to_r       #=> (2/1)
- *    '300/2'.to_r       #=> (150/1)
- *    '-9.2'.to_r        #=> (-46/5)
- *    '-9.2e2'.to_r      #=> (-920/1)
- *    '1_234_567'.to_r   #=> (1234567/1)
- *    '21 June 09'.to_r  #=> (21/1)
- *    '21/06/09'.to_r    #=> (7/2)
- *    'BWV 1079'.to_r    #=> (0/1)
- *
- * NOTE: "0.3".to_r isn't the same as 0.3.to_r.  The former is
- * equivalent to "3/10".to_r, but the latter isn't so.
+ *    str.to_r -> rational
  *
- *    "0.3".to_r == 3/10r  #=> true
- *    0.3.to_r   == 3/10r  #=> false
+ * Returns the result of interpreting leading characters in +self+ as a rational value:
```

Comment:
This new documentation loses the text about how extraneous characters are ignored. Maybe we split the examples and have another section to explain that?